src/poliastro/threebody/cr3bp_char_quant.py

```diff
@@ -0,0 +1,236 @@
+3. To non-dimensionalize a physical veloctiy vecotr, multiply by t*/l*
+4. To dimensionalize a [nd] veloctiy vecotr, multiply by l*/t*
+5. Position of P1 = [-mu, 0, 0] [nd] {x,y,z}
+5. Position of P2 = [1-mu, 0, 0] [nd] {x,y,z}
+"""
+from astropy import units as u
+
+import poliastro.constants.general as general
+import poliastro.constants.mean_elements as dist
+
+
+class SystemChars:
+    """Computes and stores the properties (mu, l*, t*) of a CR3BP system
+    'mu': mass ratio of P1-P2 primary bodies
+    'l*': characterisitic lenght of P1-P2 primary bodies
+    't*': characterisitic time of P1-P2 primary bodies
+    If P2 is more massive than P1 then swap the Pi, so that P1 is the more massive body
+    """
+
+    def __init__(self, p1="Earth", p2="Moon"):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        p1 : string
+           'body_i'. The deafult is Earth
+        p2 : string
+           'body_j'. The deafult is Moon
+        """
+
+        mu, lstar, tstar, p1, p2 = self.BodiesCharCompute(
+            p1.lower(), p2.lower()
+        )  # lower() to ensure lowercase characters
+        units_nd = u.def_unit(
+            "[nd]"
+        )  # Define custome units, [nd]: non-dimensional unit
+
+        self._mu = mu * units_nd
+        self._lstar = lstar
+        self._tstar = tstar
+        self._p1 = p1
+        self._p2 = p2
+
+    # All the attributes are made private to make them constant and avoid being mistakenly changed
+    @property
+    def mu(self):
+        """Mass ration of P1-P2 primary bodies in CR3BP"""
+        return self._mu
+
+    @property
+    def lstar(self):
+        """Characterisitc Length"""
+        return self._lstar
+
+    @property
+    def tstar(self):
+        """Characterisitc Time"""
+        return self._tstar
+
+    @property
+    def p1(self):
+        """Primary body P1"""
+        return self._p1
+
+    @property
+    def p2(self):
+        """Primary body P2"""
+        return self._p2
+
+    def __mu_calc(self, mu_pi):
+        """Calculate mu of CR3BP
+        Parameters
+        ----------
+        mu_pi : ndarray, float
+            mu_pi[0] = mu of P1
+            mu_pi[1] = mu of P2
+
+        Returns
+        -------
+        mu: float, M2/(M1+M2) = mu2/(mu1+mu2)
+            M1 and M2 are mass of Primary Bodies and M2<M1
+        """
+        return mu_pi[1] / (mu_pi[0] + mu_pi[1])
+
+    def __tstar_calc(self, mu_pi, lstar):
+        """Calculate t* of CR3BP
+        Parameters
+        ----------
+        mu_pi: ndarray, float
+            mu_pi[0] = mu of P1
+            mu_pi[1] = mu of P2
+        lstar: float, km
+            Characterisitc distance between P1 and P2
+
+        Returns
+        -------
+        t*: float, sec
+            Characterisitc time of P1-P2 system
+
+        .. math::
+
+            \sqrt{\frac{l*^3}{M1+M2}}
+        """
+        return (lstar**3 / (mu_pi[0] + mu_pi[1])) ** 0.5
```

Comment:
Let's create a new `poliastro.core.threebody` module and add `calculate_t_star` as a standalone function there.